pdns/rec-lua-conf.cc

```diff
@@ -725,11 +725,18 @@ void loadRecursorLuaConfig(const std::string& fname, luaConfigDelayedThreads& de
     lci.allowAdditionalQTypes.insert_or_assign(qtype, pair(targets, mode));
   });
 
-  Lua->writeFunction("addProxyMapping", [&proxyMapping](const string& netmaskArg, const string& addressArg) {
+  Lua->writeFunction("addProxyMapping", [&proxyMapping](const string& netmaskArg, const string& addressArg, boost::optional<std::vector<pair<int,std::string>>> smnStrings) {
     try {
       Netmask netmask(netmaskArg);
       ComboAddress address(addressArg);
-      proxyMapping.insert_or_assign(netmask, address);
+      boost::optional<SuffixMatchNode> smn;
```

Comment:
Maybe this could be `std::optional`? Not very important, but it'll be one fewer cleanup, some day.

Reply:
This ends up in a class handled by `LuaWrapper`, which has no `std::optional` support afaik.

Reply:
Ah, I didn't think that class would also hit `LuaWrapper`. In that case, indeed, carry on.